### PR TITLE
Update general.py

### LINKED ISSUE #4761: get_num_dims

### PR DESCRIPTION
**Pull request by NGStaph**
added usage example for ivy.get_num_dims
resolves #4761

### CODE REVIEW

**Comment by NGStaph** [comment]
add_reformatting_checklist_general

**Comment by ahmedo42** [comment]
Hi, @NGStaph Thanks for your contribution!!
kindly take a look at the [function formatting](https://lets-unify.ai/ivy/contributing/4_open_tasks.html#function-formatting) checklist and request a review whenever you're done.

**Comment by NGStaph** [comment]
i believe the checks which've failed aren't related to the changes that I've made. (I've just added an example for get_num_dims)

**Comment by ahmedo42** [comment]
Hi @NGStaph, the checks are passing but there are still things to implement, including the static and array instance methods, kindly check the function formatting checklist in the Deep Dive thoroughly!

**Comment by NGStaph** [comment]
I've added the examples to 'ivy/array/general.py'. If i understand correctly, 'get_num_dims 'doesn't require a container method?

**Comment by ahmedo42** [comment]
@NGStaph every function should have a container method, please read through the [function formatting](https://lets-unify.ai/ivy/contributing/4_open_tasks.html#function-formatting) carefully.

Also, discard any changes unrelated to the `get_num_dims` function

**Comment by NGStaph** [comment]
I see. In that case, i guess we'd have to reopen #4615
I'd also like to clarify that if  the function "get_num_dims" returns the number of dimensions for the container, I'd just have to implement a _container instance method_ and won't need to implement a _container static method_,right?

**Comment by ahmedo42** [comment]
@NGStaph I reopened the issue as it was not completed AFAIK.
to answer your question, the function should return the `num_dims` of the container sub-elements, so both methods should be implemented ( container AND static )

**Comment by NGStaph** [comment]
Could you review my changes and let me know if they're alright?

File: ivy/array/general.py [hunk 1]
```diff
@@ -1078,5 +1078,22 @@ def get_num_dims(self: ivy.Array, /, *, as_array: bool = False) -> int:
         ret
             Shape of the array
 
+        Examples
```
Comment (ahmedo42) [inline ivy/array/general.py:1081]: you should fix the failing docstring tests for these examples

File: ivy/functional/ivy/general.py [hunk 2]
```diff
@@ -3186,6 +3186,23 @@ def get_num_dims(
     ret
         Shape of the array
```
Comment (ahmedo42) [inline ivy/functional/ivy/general.py:3188]: you should also add the following sentence here

Both the description and the type hints above assumes an array input for simplicity,
but this function is \*nestable\*, and therefore also accepts :class:\`ivy.Container\`
instances in place of any of the arguments.

File: ivy/container/general.py [hunk 3]
```diff
@@ -3596,3 +3596,153 @@ def supports_inplace_updates(
...
+        map_sequences
+            Whether to also map method to sequences (lists, tuples). Default is False.
+        
+
+        Returns
+        -------
+        ret
+            Shape of the array
+
+        Examples
```
Comment (ahmedo42) [inline ivy/container/general.py:3639]: the static method examples should show the usage of `static_get_num_dims()` and not `ivy.get_num_dims()`

File: ivy/container/general.py [hunk 4]
```diff
@@ -3596,3 +3596,153 @@ def supports_inplace_updates(
...
+            Default is False.
+        map_sequences
+            Whether to also map method to sequences (lists, tuples). Default is False.
+        
+
+        Returns
+        -------
+        ret
+            Shape of the array
+
```
Comment (ahmedo42) [inline ivy/container/general.py:3714]: the instance method examples should show the usage of `x.get_num_dims()` where x is an `ivy.Container` , you should also add the `Examples` subheading like the other examples

**Review by ahmedo42 — CHANGES_REQUESTED** [review]

File: ivy/array/general.py (hunk 1 above)
Comment (NGStaph) [reply ivy/array/general.py:1081]: fixed

File: ivy/functional/ivy/general.py (hunk 2 above)
Comment (NGStaph) [reply ivy/functional/ivy/general.py:3188]: Added the aforementioned, Should i also add _container_ and _mixed_ examples within 'functional/ivy/general.py'?

File: ivy/container/general.py (hunk 3 above)
Comment (NGStaph) [reply ivy/container/general.py:3639]: changed as instructed

File: ivy/container/general.py (hunk 4 above)
Comment (NGStaph) [reply ivy/container/general.py:3714]: fixed

**Comment by NGStaph** [comment]
Hi, could you review the changes once again?

File: ivy/functional/ivy/general.py (hunk 2 above)
Comment (ahmedo42) [reply ivy/functional/ivy/general.py:3188]: yes go ahead and then we can merge

Comment (NGStaph) [reply ivy/functional/ivy/general.py:3188]: changed

**Review by ahmedo42 — APPROVED** [review]

**Comment by ahmedo42** [comment]
LGTM @NGStaph, Thanks!

**Pull request merged**